spec/filesystem.spec.js

```diff
@@ -232,6 +472,34 @@ describe('filesystem.spec: filesystem namespace tests', () => {
             `);
             assert.equal(runner.getOutput(), 'NE_RT_NATRTER');
         });
+
+        it('throws an error when opening a non-existent file', async () => {
+            runner.run(`
+                try {
+                    await Neutralino.filesystem.openFile(NL_PATH + '/.tmp/nonexistent.txt');
+                } catch (error) {
+                    await __close(error.code);
+                }
+            `);
+            assert.equal(runner.getOutput(), 'NE_FS_FILOPER');
+        });
+
+        it('opens a file with different modes', async () => {
+            runner.run(`
+                let fileIds = [];
+                await Neutralino.filesystem.writeFile(NL_PATH + '/.tmp/test.txt', 'Hello');
+    
+                let fileId = await Neutralino.filesystem.openFile(NL_PATH + '/.tmp/test.txt', 'read');
+                fileIds.push(fileId);
+                fileId = await Neutralino.filesystem.openFile(NL_PATH + '/.tmp/test.txt', 'write');
+                fileIds.push(fileId);
+    
+                await __close(JSON.stringify(fileIds));
+            `);
+            const fileIds = JSON.parse(runner.getOutput());
+            assert.ok(Array.isArray(fileIds));
+            assert.notEqual(fileIds[0], fileIds[1]);
+        });
```

Comment:
The `filesystem.openFile()` function only accepts one parameter (filename), so can we can turn this test case into something like "creates file identifiers with incremental integers" (or create any test case by yourself that suites here better) ?